### PR TITLE
Remove #_top anchor from search result URLs

### LINKED ISSUE #6897: Entering pages through search includes unnecessary anchor
### 📝 Issue Description

Whenever you enter a page by search (vs just clicking the sidebar link) it gets suffixed with the `#_top` anchor, which is undesired if you want to copy the link to a docs page.

### 📋 On which page(s) it occurs

Every page, and by search I mean Algolia.

### 🤔 Expected Behavior

Searching and clicking to enter a page should not include the `#_top` anchor link, e.g. https://docs.astro.build/en/install/auto/

### 👀 Current Behavior

The URL address always includes `#_top`, e.g. https://docs.astro.build/en/install/auto/#_top

### 🖥️ Browser

Chrome 120

### 📄 Additional Information

_No response_

### PR DESCRIPTION
**Pull request by TheOtterlord**
#### Description (required)

Removes the `#_top` anchor from URLs in search results.

#### Related issues & labels (optional)

- Closes #6897

### CODE REVIEW

**Review by at-the-vr — APPROVED** [review]
LGTM

**Review by liruifengv — APPROVED** [review]
LGTM

**Review by sarah11918 — APPROVED** [review]
Thank you!

File: src/components/starlight/Search.astro [hunk 1]
```diff
@@ -47,6 +47,7 @@ const docSearchStrings = getDocSearchStrings(Astro);
 				// work better on localhost, preview URLS.
 				const url = new URL(item.url);
 				if (url.hash === '#overview') url.hash = '';
```
Comment (delucis) [inline src/components/starlight/Search.astro:49]: Thanks for fixing this @TheOtterlord!

I think we could also have removed this? This was the old version of this when we used `#overview` for the top-of-page anchor, but I don't _think_ it should exist anywhere anymore?

Comment (TheOtterlord) [reply src/components/starlight/Search.astro:49]: aha, sounds good! I'll confirm this and remove it in a small follow-up PR

**Pull request merged**